client/my-sites/theme/main.jsx

```diff
@@ -791,7 +791,9 @@ class ThemeSheet extends Component {
 			plansUrl = localizeUrl( 'https://wordpress.com/pricing' );
 		} else if ( siteSlug ) {
 			const plan = isExternallyManagedTheme || isBundledSoftwareSet ? PLAN_BUSINESS : PLAN_PREMIUM;
-			plansUrl = plansUrl + `/${ siteSlug }/?plan=${ plan }`;
+			plansUrl =
+				plansUrl +
+				`/${ siteSlug }/?plan=${ plan }&redirect_to=${ encodeURIComponent( document.URL ) }`;
```

Comment:
Very likely to be just me, and it doesn't matter in the least, but i'm surprised there's not a nicer way of getting the current page. Even `window.location.href.replace( window.location.origin, '' );`. I don't think it's worth worrying over in the slightest though.

Reply:
Cannot `analyticsPath` be re-used here?

Reply:
The `analyticsPath` contains a `:site` string instead of the actual site name. I've added a hardcoded `redirectTo` const that builds the redirect URL.

The weird thing is that I've tried before a relative path and it didn't work. Apparently now it works 🪄